src/lib.rs

```diff
@@ -43,3 +43,8 @@ mod writer;
 
 // Errors should be namespaced away.
 pub mod errors;
+
+// Ensure that examples in the README are tested
+#[doc = include_str!("../README.md")]
```

Comment:
I've never seen this before. Love it.

Reply:
I found an example in the docs: https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#include-items-only-when-collecting-doctests